File: tests/fixtures/edgecase_repo/swift/Service4.swift
import Foundation

protocol Handler4 {
    func handle(_ payload: String) -> Bool
}

class Base4 {
    let name: String
    init(name: String) {
        self.name = name
    }
}

class Service4: Base4, Handler4 {
    func handle(_ payload: String) -> Bool {
        self.validate(payload)
        return true
    }

    private func validate(_ payload: String) {}
}

func helper4(_ value: String) -> String {
    return value
}

File: tests/fixtures/edgecase_repo/scala/Service4.scala
package com.example.gen

import com.example.gen.Base4

trait Handler4 {
  def handle(payload: String): Boolean
}

class Base4(val name: String)

class Service4(name: String) extends Base4(name) with Handler4 {
  def handle(payload: String): Boolean = {
    this.validate(payload)
    true
  }

  private def validate(payload: String): Unit = {}
}

object Helpers4 {
  def helper(value: String): String = value
}
